### PR TITLE
State version check

### PR DESCRIPTION
**Pull request by diegomrsantos**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java [hunk 1]
```diff
@@ -211,8 +215,25 @@ public void process(
...
+                  + " is expected to be at %s, but is at %s) - please clean the ledger and start"
+                  + " again.",
+              expectedStateVersion, this.lastStateVersionInDbOpt));
+    }
+  }
+
+  private void persistCurrentSubstateAccumulatorHash(
+      Transaction dbTxn, Optional<Long> currentStateVersion, long nextStateVersion) {
+    currentStateVersion.ifPresent(
+        it ->
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java:233]: I'd prefer a more informative name here.

**Review by siy — APPROVED** [review]
LGTM

**Review by dhedey — APPROVED** [review]

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java (hunk 1 above)
Comment (diegomrsantos) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java:233]: Repeating `currentStateVersion`? I can change it, but seems unnecessary noise in this case. What do you think?

Comment (diegomrsantos) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/substatehash/BerkeleySubstateAccumulatorHashStore.java:233]: Actually, I can't use the same name as it is already defined in the scope. I'll merge it like that for now. Please tell me what you suggest and I can change it in the next PR.

**Pull request merged**